src/lib.rs

```diff
@@ -749,15 +749,15 @@ impl AstarteSdk {
             let interface_major = interfaces
                 .interfaces
                 .get(interface_name)
-                .ok_or_else(|| {
-                    AstarteError::InterfaceError(interface::Error::InterfaceNotFoundError)
-                })?
+                .ok_or(AstarteError::InterfaceError(
```

Comment:
What kind of alarms Clippy throw here? 
Arguments passed to ok_or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use [ok_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else), which is lazily evaluated.

Reply:
The returned warning is: `clippy::unnecessary_lazy_evaluations`, description can be found [here](https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations).

The closure we are passing to `ok_or_else` is just the variant of an `enum` right?
Why would it make a difference if it's lazily or eagerly evaluated?